packages/cicero-core/src/template.js

```diff
@@ -172,14 +179,89 @@ class Template {
         return hasher.digest('hex');
     }
 
+    /**
+     * verifies the signature stored in the template object using the template hash and timestamp
+     * @return {boolean} true if signature is valid else false
+     */
+    verifyTemplateSignature() {
+        const templateHash = this.getHash();
+        if (this.authorSignature === null) {throw new Error('The template is missing author signature!');}
+        const signature = this.authorSignature.templateSignature.signature;
+        const timestamp = this.authorSignature.templateSignature.timestamp;
+        const signatoryCert = this.authorSignature.templateSignature.signatoryCert;
+        //X509 cert converted from PEM to forge type
+        const certificateForge = forge.pki.certificateFromPem(signatoryCert);
+        //public key in forge typenode index.js sign acme 123 helloworldstate
+        const publicKeyForge = certificateForge.publicKey;
+        //convert public key from forge to pem
+        const publicKeyPem = forge.pki.publicKeyToPem(publicKeyForge);
+        //convert public key in pem to public key type in node.
+        const publicKey = crypto.createPublicKey(publicKeyPem);
+        //signature verification process
+        const verify = crypto.createVerify('SHA256');
+        verify.write(templateHash + timestamp);
+        verify.end();
+        const result = verify.verify(publicKey, signature, 'hex');
+        if (!result) {
+            throw new Error('Template\'s author signature is invalid!');
+        }else{
```

Comment:
This can be removed (see previous comment) so that the method never returns a value.